source/patterns/@aws-solutions-constructs/core/lib/glue-job-defaults.ts

```diff
@@ -11,20 +11,35 @@
  *  and limitations under the License.
  */
 
-import { CfnJob, CfnJobProps } from '@aws-cdk/aws-glue';
-import { IRole } from '@aws-cdk/aws-iam';
-import { IResolvable } from '@aws-cdk/core';
+import * as glue from '@aws-cdk/aws-glue';
+import * as iam from '@aws-cdk/aws-iam';
 
-export function DefaultGlueJobProps(jobRole: IRole, jobCommand: CfnJob.JobCommandProperty | IResolvable,
-  glueSecurityConfigName: string, _defaultArguments: {}, _glueVersion: string | undefined ): CfnJobProps | any {
-  const defaultGlueJobProps: CfnJobProps = {
-    command: jobCommand,
-    role: jobRole.roleArn,
-    securityConfiguration: glueSecurityConfigName,
-    defaultArguments: _defaultArguments,
-    // glue version though optional is required for streaming etl jobs otherwise it throws an error that 'command not found'
-    ...(_glueVersion !== undefined ? { glueVersion: _glueVersion } : { glueVersion: '2.0' })
-  };
+export function DefaultGlueJobProps(jobRole: iam.IRole, userProvidedGlueJobProps: Partial<glue.CfnJobProps>,
+  glueSecurityConfigName: string, _defaultArguments: {} ): glue.CfnJobProps {
+  const _glueVersion: string | undefined = userProvidedGlueJobProps.glueVersion;
+
+  let defaultGlueJobProps: glue.CfnJobProps;
+
+  if (userProvidedGlueJobProps.workerType !== undefined && userProvidedGlueJobProps.numberOfWorkers !== undefined) {
+    defaultGlueJobProps = {
+      command: userProvidedGlueJobProps.command!,
+      role: jobRole.roleArn,
+      securityConfiguration: glueSecurityConfigName,
+      defaultArguments: _defaultArguments,
+      maxCapacity: 2, // setting default to 2 to ensure customers don't pay for something they don't need
```

Comment:
When I read this code it appears that if the user specifies workerType and numberOfWorkers, then we set maxCapacity to 2 even if the user provided one. There does not appear to be any place where a client specified maxCapacity is recognized - nor any other userProvidedGlueJobProps attributes.

Regular pattern is the service has a set of default props. When users provide props, then overrideProps is used to replace the default props with the user provided values. When there are values that _must_ have a certain value for the construct to work, then overrideProps is run a second time to override the user provided values.

We may need to have a call to discuss.